x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -28,22 +30,30 @@ import {
   EmbeddableInput,
   EmbeddableOutput,
   IContainer,
+  SavedObjectEmbeddableInput,
+  ReferenceOrValueEmbeddable,
 } from '../../../../../../src/plugins/embeddable/public';
 import { DOC_TYPE, Document, injectFilterReferences } from '../../persistence';
 import { ExpressionWrapper } from './expression_wrapper';
 import { UiActionsStart } from '../../../../../../src/plugins/ui_actions/public';
 import { isLensBrushEvent, isLensFilterEvent } from '../../types';
 
-export interface LensEmbeddableConfiguration {
-  expression: string | null;
-  savedVis: Document;
-  editUrl: string;
-  editPath: string;
-  editable: boolean;
-  indexPatterns?: IIndexPattern[];
-}
+import { IndexPatternsContract } from '../../../../../../src/plugins/data/public';
+import { getEditPath } from '../../../common';
+import { IBasePath, SavedObject } from '../../../../../../src/core/public';
+import { AttributeService } from '../../../../../../src/plugins/dashboard/public';
```

Comment:
I worry about importing dashboard-specific code into the Lens embeddable, since we also work in Canvas.